autrainer/training/training.py

```diff
@@ -800,34 +800,20 @@ def _disaggregated_evaluation(self, df, groundtruth, stratify):
         TODO: Need to define schema in the docs.
 
         """
-        df = df.set_index(groundtruth.index)
         results = {m.name: {} for m in self.data.metrics}
         for metric in self.data.metrics:
             if isinstance(self.data.target_column, list):
                 # this handles the case of multi-label classification and
                 # multi-target regression
                 # loops over all targets and computes the metric for them
-                # additionally stores avg. over all labels under "all"
-                total = 0
                 for idx, col in enumerate(self.data.target_column):
-                    if self.data.task == "ml-classification":
-                        res = metric(
-                            groundtruth[col],
-                            df["predictions"].apply(lambda x: int(col in x)),
-                        )
-                    else:
-                        res = metric(
-                            groundtruth[col],
-                            df["predictions"].apply(lambda x: x[idx]),
-                        )
-                    results[metric.name][col] = res
-                    total += res
-                total /= len(self.data.target_column)
-                results[metric.name]["all"] = total
-            else:
-                results[metric.name]["all"] = metric(
-                    groundtruth[self.data.target_column], df["predictions"]
-                )
+                    results[metric.name][col] = metric(
+                        tracker.targets[:, idx],
+                        tracker.predictions[:, idx],
+                    )
```

Comment:
> > This is a good idea, the question is if we want to create a separate function for stratified evaluation to the abstract metric.
> 
> You mean that there would be a function with the following prototype?
> 
> ```python
> def stratified(self, groundtruth: pd.DataFrame, predictions: pd.DataFrame, columns: List[str]):
> ```

Would you prefer this? I mean it works now nicely, but if a user wants to add a new multi-class / multi-target metric, they would need to perform the same checks as in `_safe_f1_score`. If we just call `metric.stratified` here it could be more clear what happens?

Reply:
Yes, I think this is cleaner. I will switch to that.

Reply:
Changed

Reply:
Yes, this looks great and is clean!